presto-spark-base/src/main/java/com/facebook/presto/spark/execution/operator/NativeExecutionOperator.java

```diff
@@ -174,13 +265,12 @@ public void noMoreSplits()
     {
     }
 
-    /*
-     * TODO: the close() method will shutdown the external process
-     */
     @Override
     public void close()
     {
         systemMemoryContext.setBytes(0);
+        task.stop();
```

Comment:
nit: add close only if they are not null as the object can be created and closed without calling start.